src/generator/properties.ts

```diff
@@ -60,7 +60,7 @@ function getFormatByDMMFType(fieldType: string): string | undefined {
 
 function getJSONSchemaForPropertyReference(field: DMMF.Field): JSONSchema7 {
     const notNullable = field.isRequired || field.isList
-    const ref = { $ref: `${DEFINITIONS_ROOT}${field.type}` }
+    const ref = { $ref: `${JSON_SCHEMA_ID}${DEFINITIONS_ROOT}${field.type}` }
```

Comment:
Is it really necessary to change the internal references to include the main $id reference? Let‘s have a look at https://json-schema.org/understanding-json-schema/structuring.html#id2. Although $id is set, it is not necessary to include $id in $refs. (bundling section)

“ You should also see that "$ref": "#/definitions/state" resolves to the definitions keyword in the address schema rather than the one at the top level schema like it would if the embedded schema wasn’t used.

Each Schema Resource is evaluated independently and may use different JSON Schema dialects. The example above has the address Schema Resource using Draft 7 while the customer Schema Resource uses Draft 2019-09. If no $schema is declared in an embedded schema, it defaults using to the dialect of the parent schema.“

Reply:
Sorry to answer just now. Got caught up by work.

I thought just adding the $id would be enough, but in case of someone using [ajv-merge-patch](https://github.com/ajv-validator/ajv-merge-patch) (like i am), if the user do not specify it the $id in the reference, the reference doesn't work as expected.
So i guess in some cases, the precision might be unnecessary/superfluous, but in other cases, it might be needed.